climate.py

```diff
@@ -321,8 +355,75 @@ def get_scouting_observation(token, api_key, scouting_observation_id):
 
     res = requests.get(uri, headers=headers)
     Logger().info(to_curl(res.request))
-    
+
     if res.status_code == 200:
         return res.json()
-    else:
-        return None
+
+    return None
+
+
+def get_scouting_observation_attachments(token,
+                                         api_key,
+                                         scouting_observation_id):
+    """
+    Retrieve attachments associated with a given scouting observation. Photos
+    added to scouting notes in the FieldView app are capped to 20MB, and we
+    won’t store photos larger than that in a scouting note.
+    https://dev.fieldview.com/technical-documentation/ for possible status
+    values and their meaning.
+    :param token: access_token
+    :param api_key: Provided by Climate
+    :param scouting_observation_id: Unique identifier of the
+        Scouting Observation.
+
+    """
+    uri = '{}/v4/layers/scoutingObservations/{}/attachments'.format(
+        api_uri, scouting_observation_id)
+    headers = {
+        'authorization': bearer_token(token),
+        'accept': json_content_type,
+        'x-api-key': api_key
+    }
+
+    res = requests.get(uri, headers=headers)
+    Logger().info(to_curl(res.request))
+
+    if res.status_code == 200:
+        return res.json()['results']
+
+    return []
+
+
+def get_scouting_observation_attachments_contents(token,
+                                                  api_key,
+                                                  scouting_observation_id,
+                                                  attachment_id):
+    """
+    Retrieve the binary contents of a scouting observation’s attachment.
+    https://dev.fieldview.com/technical-documentation/ for possible status
+    values and their meaning.
+    :param token: access_token
+    :param api_key: Provided by Climate
+    :param scouting_observation_id: Unique identifier of the Scouting
+        Observation.
+    :param attachment_id : Unique identifiler of the attachment
+
+    """
+
+    uri = '{}/v4/layers/scoutingObservations/{}/attachments/{}/contents'.\
+        format(api_uri,
+               scouting_observation_id,
+               attachment_id)
+    headers = {
+        'authorization': bearer_token(token),
+        'accept': 'image/jpeg',
+        'x-api-key': api_key,
+    }
+
+    res = requests.get(uri, headers=headers)
+    Logger().info(to_curl(res.request))
+
+    if res.status_code == 200 or res.status_code == 206:
+        return res.content
```

Comment:
We should show how to do the chunked download of an attachment, since if the image size is larger than 5MB, it must be fetched in multiple requests, and rejoined into a single file.

Reply:
added code for chunking.